source/lac/dynamic_sparsity_pattern.cc

```diff
@@ -407,6 +408,33 @@ DynamicSparsityPattern::symmetrize ()
 
 
 
+template <typename SparsityPatternTypeLeft, typename SparsityPatternTypeRight>
+void
+DynamicSparsityPattern::compute_mmult_pattern (
+  const SparsityPatternTypeLeft &left,
+  const SparsityPatternTypeRight &right)
+{
+  Assert (left.n_cols() == right.n_rows(), ExcDimensionMismatch(left.n_cols(), right.n_rows()));
+
+  this->reinit(left.n_rows(),right.n_cols());
+
+  typename SparsityPatternTypeLeft::iterator
+  it_left = left.begin(),
+  end_left = left.end();
+  for (; it_left!=end_left; ++it_left)
+    {
+      const unsigned int m = it_left->column();
+
+      typename SparsityPatternTypeRight::iterator
+      it_right = right.begin(m),
+      end_right = right.end(m);
+      for (; it_right!=end_right; ++it_right)
+        this->add(it_left->row(),it_right->column());
+    }
+}
```

Comment:
@kronbichler don't we need to call `compress` here?

Reply:
No, in a `DynamicSparsityPattern` there is no `compress`.